File: python/ex005.py
# Faça um programa que leila um número inteiro e mostre na tela o se sucessor e seu antecessor.
# usuario digita 5, antecessor é o 4 e sucessor é o 6
number = int(input("Digite um numero: "))
print("O antecessor do numero {} é {} e o sucessor é {}".format(number, number - 1, number + 1))

File: rust/ex005.rs
use std::io;

fn main() {
    let mut number = String::new();
    println!("Digite um numero: ");

    io::stdin()
        .read_line(&mut number)
        .expect("Erro ao ler número, tente novamente.");

    // ::<> Turbofish
    match number.trim().parse::<i32>() {
        Ok(num) => {
            println!(
                "O antecessor do numero {:#?} é {:#?} e o sucessor é {:#?}",
                num,
                num - 1,
                num + 1
            );
        }
        Err(_) => {
            println!("Por favor, digite um número válido!");
        }
    }

}
